GVRf/Framework/src/org/gearvrf/GVRActivity.java

```diff
@@ -319,12 +319,17 @@ boolean updateSensoredScene() {
      * @param view Is a {@link GVRView} that draw itself into some
      *            {@link GVRViewSceneObject}.
      */
-    public void registerView(View view) {
+    public void registerView(final View view) {
         /* The full screen should be updated
            otherwise just the children's bounds may be refreshed. */
         mRenderableViewGroup.setClipChildren(false);
```

Comment:
When in doubt, make the android view calls on the main thread. Since we don't really know on which thread registerView may be called, I vote for the main thread.